sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp

```diff
@@ -55,405 +54,484 @@ namespace Azure { namespace Core { namespace Http {
+       * @param bufferSize size of the buffer to send.
+       * @return CURL_OK when response is sent successfully.
+       */
+      CURLcode SendBuffer(uint8_t const* buffer, size_t bufferSize);
 
-    /**
-     * @brief This function is used after sending an HTTP request to the server to read the HTTP
-     * RawResponse from wire until the end of headers only.
-     *
-     * @return CURL_OK when an HTTP response is created.
-     */
-    void ReadStatusLineAndHeadersFromRawResponse();
+      /**
+       * @brief This function is used after sending an HTTP request to the server to read the HTTP
+       * RawResponse from wire until the end of headers only.
+       *
+       * @return CURL_OK when an HTTP response is created.
+       */
+      void ReadStatusLineAndHeadersFromRawResponse();
 
-    /**
-     * @brief Reads from inner buffer or from Wire until chunkSize is parsed and converted to
-     * unsigned long long
-     *
-     */
-    void ParseChunkSize();
+      /**
+       * @brief Reads from inner buffer or from Wire until chunkSize is parsed and converted to
+       * unsigned long long
+       *
+       */
+      void ParseChunkSize();
 
-    /**
-     * @brief This function is used when working with streams to pull more data from the wire.
-     * Function will try to keep pulling data from socket until the buffer is all written or until
-     * there is no more data to get from the socket.
-     *
-     * @param buffer ptr to buffer where to copy bytes from socket.
-     * @param bufferSize size of the buffer and the requested bytes to be pulled from wire.
-     * @return return the numbers of bytes pulled from socket. It can be less than what it was
-     * requested.
-     */
-    int64_t ReadSocketToBuffer(uint8_t* buffer, int64_t bufferSize);
+      /**
+       * @brief This function is used when working with streams to pull more data from the wire.
+       * Function will try to keep pulling data from socket until the buffer is all written or until
+       * there is no more data to get from the socket.
+       *
+       * @param buffer ptr to buffer where to copy bytes from socket.
+       * @param bufferSize size of the buffer and the requested bytes to be pulled from wire.
+       * @return return the numbers of bytes pulled from socket. It can be less than what it was
+       * requested.
+       */
+      int64_t ReadSocketToBuffer(uint8_t* buffer, int64_t bufferSize);
 
-    bool IsEOF()
-    {
-      return this->m_isChunkedResponseType ? this->m_chunkSize == 0
-                                           : this->m_contentLength == this->m_sessionTotalRead;
-    }
+      bool IsEOF()
+      {
+        return this->m_isChunkedResponseType ? this->m_chunkSize == 0
+                                             : this->m_contentLength == this->m_sessionTotalRead;
+      }
 
-  public:
+    public:
 #ifdef TESTING_BUILD
-    // Makes possible to know the number of current connections in the connection pool
```

Comment:
Can we get rid of these and instead make the Testclass a friend when TESTING_BUILD is enabled?